sanic/server/websockets/connection.py

```diff
@@ -47,7 +49,13 @@ async def recv(self, *args, **kwargs) -> Optional[str]:
         message = await self._receive()
 
         if message["type"] == "websocket.receive":
-            return message["text"]
+            try:
+                return message["text"]
+            except KeyError:
+                try:
+                    return message["bytes"].decode()
```

Comment:
Shouldn't bytes be received as is, without decoding? This is what the built-in server does, to support binary data.

Reply:
I wanted to receive it for now without breaking the existing contract. I think we need to revisit this.

Reply:
Yes, we need to. I would think this being necessary for 23.12.1 release because we cannot have LTS processing it wrong like that. Apparently not many use ASGI mode with binary WebSockets because this wasn't caught earlier.

The binary format is specifically meant for binary data, for which my own use cases include MsgPack, audio streaming and other raw data that will never be able to be decoded into text (if not for a freak accident of chance that still breaks the software). Never JSON which can and should be sent in text mode instead.

Reply:
Yeah, probably sooner than a year from now and back port it.